crates/common/src/opa.rs

```diff
@@ -280,6 +292,106 @@ impl PolicyLoader for CliPolicyLoader {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct UrlPolicyLoader {
+    policy_id: String,
+    address: String,
+    policy: Vec<u8>,
+    entrypoint: String,
+}
+
+impl UrlPolicyLoader {
+    pub fn new(url: &str, policy_id: &str, entrypoint: &str) -> Self {
+        Self {
+            address: url.into(),
+            policy_id: policy_id.to_owned(),
+            entrypoint: entrypoint.to_owned(),
+            ..Default::default()
+        }
+    }
+
+    async fn fetch_bundle_from_url(&mut self) -> Result<Vec<u8>, PolicyLoaderError> {
+        let response = reqwest::get(&self.address).await?;
+        if response.status().is_success() {
+            let bytes = response.bytes().await?;
+            Ok(bytes.to_vec())
+        } else {
+            Err(PolicyLoaderError::HttpError(response.status().to_string()))
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl PolicyLoader for UrlPolicyLoader {
+    fn set_address(&mut self, address: &str) {
+        self.address = address.to_owned();
+    }
+
+    fn set_rule_name(&mut self, name: &str) {
+        self.policy_id = name.to_owned();
+    }
+
+    fn set_entrypoint(&mut self, entrypoint: &str) {
+        self.entrypoint = entrypoint.to_owned();
+    }
+
+    fn get_address(&self) -> &str {
+        &self.address
+    }
+
+    fn get_rule_name(&self) -> &str {
+        &self.policy_id
+    }
+
+    fn get_entrypoint(&self) -> &str {
+        &self.entrypoint
+    }
+
+    fn get_policy(&self) -> &[u8] {
+        &self.policy
+    }
+
+    fn load_policy_from_bytes(&mut self, policy: &[u8]) {
+        self.policy = policy.to_vec();
+    }
+
+    async fn load_policy(&mut self) -> Result<(), PolicyLoaderError> {
+        let address = &self.address;
+        let bundle = match Url::parse(address) {
+            Ok(url) if url.scheme() == "file" => {
+                let file_path = url
+                    .to_file_path()
+                    .map_err(|_| PolicyLoaderError::InvalidPath(address.to_string()))?;
+                std::fs::read(file_path)?
+            }
+            Ok(_) => self.fetch_bundle_from_url().await?,
+            Err(_) => {
+                let path = Path::new(address);
+                if path.is_file() {
+                    std::fs::read(path)?
+                } else {
+                    return Err(PolicyLoaderError::InvalidPath(address.to_string()));
+                }
+            }
+        };
+
+        info!(loaded_policy_bytes=?bundle.len(), "Loaded policy");
```

Comment:
Seems odd to log "loaded policy" before calling `load_policy_from_bundle`.
```suggestion
        info!(fetched_policy_bytes=?bundle.len(), "Fetched policy");
```